apistructs/pipeline_task.go

```diff
@@ -31,6 +31,19 @@ const (
 	PipelineTaskMaxRetryDuration = 24 * time.Hour
 )
 
+type TaskParamSource string
+
+const (
+	DefaultTaskParamSource TaskParamSource = "default"
+	UserTaskParamSource    TaskParamSource = "user"
+)
+
+type TaskParamDetail struct {
+	Name   string          `json:"name"`
+	Value  string          `json:"value"`
+	Source TaskParamSource `json:"source"`
```

Comment:
type TaskParamDetail struct {
        Name
        Values map[TaskParamSource]string
}

values map should have three source: default, user and merged.